google-api-client/pom.xml

```diff
@@ -125,29 +125,41 @@
       <groupId>com.google.http-client</groupId>
       <artifactId>google-http-client-gson</artifactId>
     </dependency>
+    <dependency>
+      <groupId>com.google.guava</groupId>
+      <artifactId>guava</artifactId>
+    </dependency>
     <dependency>
       <groupId>com.google.http-client</groupId>
-      <artifactId>google-http-client-protobuf</artifactId>
-      <scope>test</scope>
+      <artifactId>google-http-client-apache-v2</artifactId>
     </dependency>
     <dependency>
-      <groupId>junit</groupId>
-      <artifactId>junit</artifactId>
-      <scope>test</scope>
+      <groupId>org.apache.httpcomponents</groupId>
+      <artifactId>httpcore</artifactId>
     </dependency>
     <dependency>
-      <groupId>commons-codec</groupId>
-      <artifactId>commons-codec</artifactId>
-      <scope>provided</scope>
-      <version>1.15</version>
-   </dependency>
```

Comment:
Git blame showed this dependency was added by the change below. This does not tell why it was added.

> [OAuth Issue 34] Testing interface for System.currentTimeMillis() http://codereview.appspot.com/6112058/


Here is the dependency tree for this PR. It picks `commons-codec:commons-codec:jar:1.11` through `org.apache.httpcomponents:httpclient:jar:4.5.13`.

```
[INFO] --- maven-dependency-plugin:3.1.2:tree (default-cli) @ google-api-client ---
[INFO] com.google.api-client:google-api-client:jar:1.31.2
[INFO] +- com.google.oauth-client:google-oauth-client:jar:1.31.4:compile
[INFO] +- com.google.http-client:google-http-client-gson:jar:1.38.1:compile
[INFO] |  \- com.google.code.gson:gson:jar:2.8.6:compile
[INFO] +- com.google.guava:guava:jar:30.1-android:compile
[INFO] |  +- com.google.guava:failureaccess:jar:1.0.1:compile
[INFO] |  +- com.google.guava:listenablefuture:jar:9999.0-empty-to-avoid-conflict-with-guava:compile
[INFO] |  +- com.google.code.findbugs:jsr305:jar:3.0.2:compile
[INFO] |  +- org.checkerframework:checker-compat-qual:jar:2.5.5:compile
[INFO] |  +- com.google.errorprone:error_prone_annotations:jar:2.3.4:compile
[INFO] |  \- com.google.j2objc:j2objc-annotations:jar:1.3:compile
[INFO] +- com.google.http-client:google-http-client-apache-v2:jar:1.38.1:compile
[INFO] +- org.apache.httpcomponents:httpcore:jar:4.4.14:compile
[INFO] +- org.apache.httpcomponents:httpclient:jar:4.5.13:compile
[INFO] |  +- commons-logging:commons-logging:jar:1.2:compile
[INFO] |  \- commons-codec:commons-codec:jar:1.11:compile
[INFO] +- com.google.http-client:google-http-client:jar:1.38.1:compile
[INFO] |  +- io.opencensus:opencensus-api:jar:0.24.0:compile
[INFO] |  |  \- io.grpc:grpc-context:jar:1.22.1:compile
[INFO] |  \- io.opencensus:opencensus-contrib-http-util:jar:0.24.0:compile
[INFO] +- com.google.protobuf:protobuf-java:jar:3.14.0:test
[INFO] +- com.google.http-client:google-http-client-protobuf:jar:1.38.1:test
[INFO] \- junit:junit:jar:4.13.1:test
[INFO]    \- org.hamcrest:hamcrest-core:jar:1.3:test
```

Reply:
Todo: does the commons-codec 1.11 have known vulnerabilities?

Reply:
I found b/172238231 tells it should be more than or equal to 1.14.

Reply:
Initially I though that the commons-codec was declared here so that the library users would have the newer version of commons-coded than the one `org.apache.httpcomponents:httpclient` draws.

However, even before this PR, it had "provided" scope. The provided dependency does not have any effect to library users' project. Therefore, there shouldn't be any effect in removing this commons-codec dependency from this pom.xml.